engine/crates/federated-graph/src/federated_graph.rs

```diff
@@ -9,7 +9,7 @@
 /// Does not guarantee:
 ///
 /// - The ordering of items inside each `Vec`.
-#[derive(serde::Serialize, serde::Deserialize)]
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
```

Comment:
nitpicking that may be irrelevant since I haven't read the rest of the diff yet, but an `Arc<FederatedGraph>` would be cheaper to clone than what I'm assuming is cloning the entire data structure here.

Reply:
Yeah, I considered that.  But the engine wants to take an owned FederatedGraph so we'd still need to be able to clone the contents.